synapse/crypto/keyring.py

```diff
@@ -667,28 +667,36 @@ async def get_server_verify_key_v2_indirect(
             perspective_name,
         )
 
+        request: JsonDict = {}
+        for queue_value in keys_to_fetch:
+            # there may be multiple requests for each server, so we have to merge
+            # them intelligently.
+            request_for_server = {
+                key_id: {
+                    "minimum_valid_until_ts": queue_value.minimum_valid_until_ts,
+                }
+                for key_id in queue_value.key_ids
+            }
+            request.setdefault(queue_value.server_name, {}).update(request_for_server)
```

Comment:
I haven't seen this idiom used much (instead of e.g. `defaultdict(dict)`),  but it a) makes sense and b) seems to be used across Synapse.